2008-05-14  Ulrich Drepper  <drepper@redhat.com>

	[BZ #6461]
	* iconv/gconv_simple.c (BODY for __gconv_transform_ascii_internal):
	Add missing braces.
	(BODY for __gconv_transform_internal_ascii): Likewise.

--- libc/iconv/gconv_simple.c	12 Oct 2007 04:40:33 -0000	1.67
+++ libc/iconv/gconv_simple.c	14 May 2008 22:52:44 -0000	1.68
@@ -820,9 +820,11 @@ ucs4le_internal_loop_single (struct __gc
 	STANDARD_FROM_LOOP_ERR_HANDLER (1);				      \
       }									      \
     else								      \
-      /* It's an one byte sequence.  */					      \
-      *((uint32_t *) outptr) = *inptr++;				      \
-      outptr += sizeof (uint32_t);					      \
+      {									      \
+	/* It's an one byte sequence.  */				      \
+	*((uint32_t *) outptr) = *inptr++;				      \
+	outptr += sizeof (uint32_t);					      \
+      }									      \
   }
 #define LOOP_NEED_FLAGS
 #include <iconv/loop.c>
@@ -851,9 +853,11 @@ ucs4le_internal_loop_single (struct __gc
 	STANDARD_TO_LOOP_ERR_HANDLER (4);				      \
       }									      \
     else								      \
-      /* It's an one byte sequence.  */					      \
-      *outptr++ = *((const uint32_t *) inptr);				      \
-      inptr += sizeof (uint32_t);					      \
+      {									      \
+	/* It's an one byte sequence.  */				      \
+	*outptr++ = *((const uint32_t *) inptr);			      \
+	inptr += sizeof (uint32_t);					      \
+      }									      \
   }
 #define LOOP_NEED_FLAGS
 #include <iconv/loop.c>
